packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -38,48 +54,56 @@ class CodeSizeController {
     _diffRoot.value = newRoot;
   }
 
-  /// The active diff tree type used to build the diff treemap.
-  ValueListenable<DiffTreeType> get activeDiffTreeType {
-    return _activeDiffTreeTypeNotifier;
+  void clearDiff() {
+    _diffRoot.value = null;
+    clearOldDiffSnapshotFile();
+    clearNewDiffSnapshotFile();
   }
 
-  // TODO(peterdjlee): Cache each diff tree so that it's less expensive
-  //                   to change bettween diff tree types.
-  void changeActiveDiffTreeType(DiffTreeType newDiffTreeType) {
-    _activeDiffTreeTypeNotifier.value = newDiffTreeType;
-    loadFakeDiffTree(
-      _diffOldSnapshotFile.value,
-      _diffNewSnapshotFile.value,
-      diffTreeType: newDiffTreeType,
-    );
+  ValueListenable<String> get oldDiffSnapshotFile => _oldDiffSnapshotFile;
+  final _oldDiffSnapshotFile = ValueNotifier<String>(null);
+
+  void loadOldDiffSnapshotFile(String filePath) {
```

Comment:
rename to changeOldDiffSnapshotFilePath, as we are not actually loading a file, but rather setting a path var. Same for new diff and snapshot